crates/burn-core/src/optim/decay.rs

```diff
@@ -21,11 +21,11 @@ pub struct WeightDecayState<B: Backend, const D: usize> {
 
 /// Weight decay implementation that transforms gradients.
 #[derive(Clone)]
-pub struct WeightDecay<B: Backend> {
-    penalty: B::FloatElem,
+pub struct WeightDecay {
+    penalty: f32,
```

Comment:
Makes sense! For uniformity, might as well change the `WeightDecayConfig` penalty to f32 as well.

And then in `WeightDecay::new(...)` we don't need to call `.elem()`.

Reply:
Oh yes nice catch :)